volatility3/framework/layers/qemu.py

```diff
@@ -192,6 +235,63 @@ def _load_segments(self):
             else:
                 raise exceptions.LayerException(self._name, f'QEMU unknown section encountered: {section_byte}')
 
+        # If no architecture has been set, try to determine it using fallback mechanisms
+        if not self._architecture:
+            self._architecture = self._fallback_determine_architecture()
+            if self._architecture is None:
+                vollog.log(constants.LOGLEVEL_VV, f"QEVM architecture could not be determined")
+
+        # Once all segments have been read, determine the PCI hole if any
+        for regex in self.pci_hole_table:
+            if regex.match(self._architecture):
+                self._pci_hole_minimum, self._pci_hole_start, self._pci_hole_end = self.pci_hole_table[regex]
+                if self.maximum_address < self._pci_hole_minimum:
+                    # The PCI hole isn't present because we're below the minimum value
+                    self._pci_hole_start, self._pci_hole_end = 0, 0
+                vollog.log(constants.LOGLEVEL_VVVV, f"QEVM architecture detected as: {self._architecture}")
+                break
+        else:
+            vollog.log(constants.LOGLEVEL_VVVV, f"QEVM unknown architecture found: {self._architecture}")
+
+    def _fallback_determine_architecture(self) -> str:
+        architecture_pattern = rb'pc-(i440fx|q35)-([0-9]{1,2}.[0-9]{1,2}(?:.[0-9]{1,2})?)'
```

Comment:
I noticed that this should include support for Ubuntu/RHEL-releases as well. Suggestion: `rb'pc-(i440fx|q35)-(\d{1,2}.\d{1,2}|[\w\d\.]+)'`. I do not exactly remember why I put `(?:.[0-9]{1,2})?)` in the last part of the expression.